src/FwdState.cc

```diff
@@ -458,7 +458,7 @@ FwdState::useDestinations()
 void
 FwdState::fail(ErrorState * errorState)
 {
-    debugs(17, 3, err_type_str[errorState->type] << " \"" << Http::StatusCodeString(errorState->httpStatus) << "\"\n\t" << entry->url());
+    debugs(17, 3, errorState << "; was: " << err);
```

Comment:
This change reduces code duplication, adds more debugging information for complex FwdState retries, synchronizes affected FwdState and TunnelStateData code, and also removes unnecessary (and excessive) multiline reporting for a level-3 debug message that complicates extracting kid-specific cache.log lines.